app/bindings/api/v0/swagger.json

```diff
@@ -325,18 +341,46 @@
           "type": "integer",
           "description": "The event's numerical order within this session. E.g. the first event after a session is started should give 0 for this field, the next one should give 1, etc."
         },
-        "page_id": {
+        "target_id": {
           "type": "string",
-          "description": "The page id."
+          "description": "The target element id."
         },
-        "book_id": {
+        "target_type": {
           "type": "string",
-          "description": "The book id."
+          "description": "The target element (interactive) type."
+        },
+        "target_attributes": {
+          "type": "string",
+          "description": "The target element attributes (map, all attributes)."
+        },
+        "context_id": {
+          "type": "string",
+          "description": "The context element id for the target element (likely a parent: eg: if target is a solution, context would be problem)."
+        },
+        "context_type": {
+          "type": "string",
+          "description": "The context element (parent) type."
+        },
+        "context_attributes": {
+          "type": "string",
+          "description": "The context element attributes (map, all useful attributes)."
+        },
+        "context_region": {
+          "type": "string",
+          "description": "The UX region the context element is in (e.g. toc, header, page)."
+        },
+        "context_state_change": {
+          "type": "string",
+          "description": "The context element state change that triggered event, if any."
```

Comment:
this is kinda pedantic, but i would say it isn't the context element that is changing state, but the target element, the name and description here are potentially confusing

Reply:
done